### PR TITLE
Resolve: #4 + Spark RDD example

### PR DESCRIPTION
**Pull request by icharo-tb**
I deleted the whole repository and cloned it again, therefore, I have no Hadoop commits saved anymore, but since they are already on the original repo, its ok. I changed this 2 Spark files again, set them up in a new branch called "spark_rdd".

Hope that everything goes neat now. I will update the repo everytime from now on to avoid having this kind of problems again.
If there is something you'd like me to change, just tell me! :3

### CODE REVIEW

File: Tools/Apache Spark.md [hunk 1]
```diff
@@ -20,7 +20,30 @@ https://spark.apache.org/docs/latest/
 
 ## Apache Spark Disadvantages
 
-#placeholder/description 
+- Spark lacks a native storage option
+- Wrong usage of RDD partitions on Spark Context can cause negative effects over HDFS and MemoryOverhead driver
+
+# Apache Spark storage
```
Comment (JPHaus) [inline Tools/Apache Spark.md:26]:
```suggestion
## Apache Spark storage
```

File: Tools/Apache Spark.md [hunk 2]
```diff
@@ -20,7 +20,30 @@ https://spark.apache.org/docs/latest/
...
+
+Apache Spark is compatible with Hadoop APIs, like HDFS. Spark also works with other storage services such as NoSQL databases, ElasticSearch and Amazon S3.
+
+# Apache Spark model
```
Comment (JPHaus) [inline Tools/Apache Spark.md:30]:
```suggestion
## Apache Spark model
```

File: Tools/Apache Spark.md [hunk 3]
```diff
@@ -20,7 +20,30 @@ https://spark.apache.org/docs/latest/
...
+
+Spark describes tasks based on a DAG (**Directed Acyclic Graphs**), it gives programmers the option of developing complex pipelines. To understand what DAG does, we must focus on its name:
+
+- *Directed*: stands for a one way process
+- *Acyclic*: means that there are no loops on the tasks
+- *Graph*: is a reference on how they can be displayed as an actual graph
+
+DAG system is also fault tolerant.
+
+# Apache Spark RDDs
```
Comment (JPHaus) [inline Tools/Apache Spark.md:42]:
```suggestion
## Apache Spark RDDs
```

File: Tools/Apache Spark.md [hunk 4]
```diff
@@ -20,7 +20,30 @@ https://spark.apache.org/docs/latest/
...
+
+Apache Spark's programming model is based on **parallel operators**. This is Spark's main advantage and feature, which let us use a Leader-Follower strategy to tackle data.
+
+Spark describes tasks based on a DAG (**Directed Acyclic Graphs**), it gives programmers the option of developing complex pipelines. To understand what DAG does, we must focus on its name:
+
+- *Directed*: stands for a one way process
+- *Acyclic*: means that there are no loops on the tasks
+- *Graph*: is a reference on how they can be displayed as an actual graph
+
+DAG system is also fault tolerant.
```
Comment (JPHaus) [inline Tools/Apache Spark.md:40]:
```suggestion
A DAG system is also fault-tolerant.
```

File: Tools/Apache Spark.md [hunk 5]
```diff
@@ -20,7 +20,30 @@ https://spark.apache.org/docs/latest/
...
+
+## Apache Spark RDDs
+
+RDDs (**Resilient Distributed Datasets**) are the main Spark abstraction. They consist of element sets that are fault tolerance and able to be parallel processed. RDDs also provide scalability due to being distributed processes as well as being immutable. 
+
+RDDs admit two kind of operations: **transformations** (such as filter, map...) and **actions** (such as reduce, collect...). Spark RDDs tend to perform lazy evaluation, which improves efficiency by executing operations only when they are needed.
```
Comment (JPHaus) [inline Tools/Apache Spark.md:46]:
```suggestion
RDDs emit two kind of operations: **transformations** (such as filter, map...) and **actions** (such as reduce, collect...). Spark RDDs tend to perform lazy evaluation, which improves efficiency by executing operations only when they are needed.
```
Not sure if you meant emit or some other word here.

File: Tutorials/Apache Spark RDD example.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+---
+Aliases: [Apache Spark]
```
Comment (JPHaus) [inline Tutorials/Apache Spark RDD example.md:2]:
```suggestion
Aliases: []
```

File: Tutorials/Apache Spark RDD example.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+Tags: [seedling]
+publish: true
+---
+
+# Overview
+
+This tutorial will cover basic RDD operations that can be run on either Google Colab or Databricks Community Edition.
+
+# Official Documentation
```
Comment (JPHaus) [inline Tutorials/Apache Spark RDD example.md:11]:
```suggestion
## Official Documentation
```
Besides the overview, all of the other headings should be h2 or lower.

File: Tools/Apache Spark.md (hunk 5 above)
Comment (icharo-tb) [reply Tools/Apache Spark.md:46]: I meant that, as a default option, RDDs take lists and dictionaries. Other datatypes must be set up

File: Tutorials/Apache Spark RDD example.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+https://spark.apache.org/docs/latest/rdd-programming-guide.html
+
+# Configuration
```
Comment (JPHaus) [inline Tutorials/Apache Spark RDD example.md:15]:
```suggestion
## Configuration
```

File: Tutorials/Apache Spark RDD example.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+After the module is installed, we will set up Spark Configuration so we can use **SparkContext**:
+
+```
+from pyspark import SparkContext, SparkConf
+
+conf = SparkConf().setAppName('test').setMaster('local')
+sc = SparkContext(conf=conf)
+```
+
+# First Steps
```
Comment (JPHaus) [inline Tutorials/Apache Spark RDD example.md:34]:
```suggestion
## First Steps
```

File: Tools/Apache Spark.md (hunk 5 above)
Comment (icharo-tb) [reply Tools/Apache Spark.md:46]: Sorry, wrong answer. I meant there that RDDs are able to execute two types of operations: transformations and actions

File: Tutorials/Apache Spark RDD example.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+```
+nums2 = sc.parallelize([3,2,1,4,5])
+evens = nums2.filter(lambda elem: elem%2==0)
+odds = nums2.filter(lambda elem: elem%2!=0)
+
+order = pairs.union(impairs)
+order.takeOrdered(5)
+```
+```
+[1, 2, 3, 4, 5]
```
Comment (JPHaus) [inline Tutorials/Apache Spark RDD example.md:49]: Great example and explantation!

File: Tools/Apache Spark.md (hunk 5 above)
Comment (icharo-tb) [reply Tools/Apache Spark.md:46]: So basicly, you execute a filtrr over an RDD that is a transformation, and later on you execute a collect, which is an action, to "print" the RDD content

**Review by JPHaus — APPROVED** [review]
Thank you again for the contributions - I hope you don't mind the minor formatting edit I made.

**Comment by icharo-tb** [comment]
No problem at all! I'm okay with the changes so far :)

**Pull request merged**